### PR TITLE
GH-799 Improve dependency loader performance with parallel downloading and relocation.

### PR DESCRIPTION
**Pull request by Rollczi**
Resolve: #799

# Tests
A - Rollczi's computer 6x 4.90GHz, 48GB, 40Mbps
B - vLucky's computer 12Mbps
C - VPS 4x 3,7 GHz, 8GB, 730Mbps

### Before
A - 90s (38s + 52s)
B - 98s (45s + 53s)
C - 54s (19s + 35s) 

### After
A - 21s (15s + 6s)
B - 31s (18s + 13s)
C - 12s (10s + 2s)

### Results
A - **4,2x** faster
B - **3,2x** faster
C - **4,5x** faster

(first part of the time represents searching for dependency, second represents downloading and relocation)

### CODE REVIEW

File: eternalcore-plugin/build.gradle.kts [hunk 1]
```diff
@@ -32,6 +32,10 @@ eternalShadowCompiler {
     }
 }
 
+dependencies {
+    implementation("com.spotify:completable-futures:0.3.6")
```
Comment (vLuckyyy) [inline eternalcore-plugin/build.gradle.kts:36]: Move to Versions.kt Constant file.

**Review by vLuckyyy — APPROVED** [review]

**Review by Jakubk15 — APPROVED** [review]

**Review by imDMK — APPROVED** [review]

File: buildSrc/src/main/kotlin/eternalcore-repositories.gradle.kts [hunk 2]
```diff
@@ -5,11 +5,8 @@ plugins {
 repositories {
     mavenCentral()
     maven { url = uri("https://hub.spigotmc.org/nexus/content/repositories/snapshots/") }
-    maven { url = uri("https://repo.panda-lang.org/releases/") }
     maven { url = uri("https://repo.papermc.io/repository/maven-public/") }
-    maven { url = uri("https://repository.minecodes.pl/releases/") }
-    maven { url = uri("https://repository.minecodes.pl/snapshots/") }
-    maven { url = uri("https://repo.eternalcode.pl/snapshots/") }
+    maven { url = uri("https://repo.panda-lang.org/releases/") }
```
Comment (CitralFlo) [inline buildSrc/src/main/kotlin/eternalcore-repositories.gradle.kts:9]: repositories could be sorted alphabetically

**Review by CitralFlo — APPROVED** [review]

File: buildSrc/src/main/kotlin/eternalcore-repositories.gradle.kts (hunk 2 above)
Comment (Rollczi) [reply buildSrc/src/main/kotlin/eternalcore-repositories.gradle.kts:9]: Alphabetically sorting doesn't make sense (also for dependencies or other things like as fields). Currently we sort repositories by frequency of use (by dependencies) to reduce dependency resolving time.

Comment (CitralFlo) [reply buildSrc/src/main/kotlin/eternalcore-repositories.gradle.kts:9]: Yessirski

**Pull request merged**